Init build-repository before publishing in CD

lorenyu opened the pull request:
## Ticket

Resolves #180 

## Changes
* Init build-repository module before publishing
* Fix log message in deploy script
* Fix directory that terraform init runs in as part of deploy script
* Disable cd in template and add instructions for enabling

## Context for reviewers
CD was failing to publish the built docker image since the build-repository module wasn't initialized. See [this run](https://github.com/navapbc/platform-test/actions/runs/3480941215/jobs/5821384944) as an example. This PR runs terraform init first as part of release-publish.

## Testing
CI

danielleswyhart commented:
I'd rather see the account ID in the output at the end of a deploy instead of the image tag which doesnt really help me to know which account actually received that deploy. Since the account ID is driven by how you configured things and you could very well be deploying to many different accounts, it would nice to know which account was just deployed to. 

either way you choose, it LGTM.

lorenyu commented:
I think the account id being a long string of numbers is not as useful as the environment name, so I'm going to stick with the environment name which I think is more human readable than the account id. The account id can still be found in the logs a few lines above, it only takes an extra two seconds to scroll up to find it.

danielleswyhart commented:
So long as the account ID is visible when the deploy is complete and not having to dig to find out which account your env might have been mistakenly configured to connect to. Hopefully, if someone is using multiple accounts they will put other safeguards into their terraform code to prevent accidental deploys to the wrong account ID.

lorenyu commented:
Sounds good. Checking which account the env is configured to is a good idea that I can add to the backlog to give project teams a way to verify that. That should be a one-time thing rather than something that they verify on every deploy.

btw I added just one more commit to disable cd in the template since cd isn't going to work until the dev environment is set up. And I added instructions for enabling it once the app environment is set up. I'll go ahead and merge now but let me know if you think there are any needed changes I can do them post merge in another PR

danielleswyhart commented:
Don't forget that we need to get the instructions updated to include the step of getting the image tag for the first deploy.

lorenyu commented:
Ah yes, that's ticket #168 , I'll get on that thanks for the reminder
